core/kamon-core-tests/src/test/scala/kamon/trace/W3CTraceContextSpanPropagationSpec.scala

```diff
@@ -0,0 +1,75 @@
+package kamon.trace
+
+import kamon.context.{Context, HttpPropagation}
+import kamon.trace.Trace.SamplingDecision
+import org.scalatest.{Matchers, OptionValues, WordSpecLike}
+
+import scala.collection.mutable
+
+class W3CTraceContextSpanPropagationSpec extends WordSpecLike with Matchers with OptionValues {
+  val traceContextPropagation = SpanPropagation.W3CTraceContext()
+
+  "The TraceContext Span propagation for HTTP" should {
+    "write the Span data into headers" in {
+      val headersMap = mutable.Map.empty[String, String]
+      traceContextPropagation.write(testContext(), headerWriterFromMap(headersMap))
+
+      headersMap.get("traceparent").value shouldBe "00-00000000000000000000000001020304-0000000004030201-01"
+      headersMap.get("tracestate").value shouldBe ""
+    }
+
+    "not inject anything if there is no Span in the Context" in {
+      val headersMap = mutable.Map.empty[String, String]
+      traceContextPropagation.write(Context.Empty, headerWriterFromMap(headersMap))
+      headersMap.values shouldBe empty
+    }
+
+    "extract a RemoteSpan from incoming headers when all fields are set" in {
+      val headersMap = Map(
+        "traceparent" -> "00-00000000000000000000000001020304-0000000004030201-01",
+        "tracestate" -> "2222"
+      )
+
+      val spanContext = traceContextPropagation.read(headerReaderFromMap(headersMap), Context.Empty).get(Span.Key)
+      spanContext.id.string shouldBe "00000000000000000000000001020304"
+      spanContext.parentId.string shouldBe "0000000004030201"
+      spanContext.trace.id.string shouldBe "00000000000000000000000001020304"
+      spanContext.trace.samplingDecision shouldBe SamplingDecision.Sample
+    }
+  }
+
+  def headerWriterFromMap(map: mutable.Map[String, String]): HttpPropagation.HeaderWriter = new HttpPropagation.HeaderWriter {
+    override def write(header: String, value: String): Unit = map.put(header, value)
+  }
+
+  def headerReaderFromMap(map: Map[String, String]): HttpPropagation.HeaderReader = new HttpPropagation.HeaderReader {
+    override def read(header: String): Option[String] = {
+      if(map.contains("fail"))
+        sys.error("failing on purpose")
+
+      map.get(header)
+    }
+
+    override def readAll(): Map[String, String] = map
+  }
+
+  def testContext(): Context =
+    Context.of(Span.Key, Span.Remote(
+      id = Identifier("4321", Array[Byte](4, 3, 2, 1)),
+      parentId = Identifier("2222", Array[Byte](2, 2, 2, 2)),
+      trace = Trace(
+        id = Identifier("1234", Array[Byte](1, 2, 3, 4)),
+        samplingDecision = SamplingDecision.Sample
```

Comment:
Should we add a test for the `Unknown` sampling decision?

Reply:
@ivantopo when we implement support for OTel trace exporter the [trace_id](https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry/proto/trace/v1/trace.proto#L66) HAS to be 16 bytes.
```
  // A unique identifier for a trace. All spans from the same trace share
  // the same `trace_id`. The ID is a 16-byte array. An ID with all zeroes
  // is considered invalid.
  //
  // This field is semantically required. Receiver should generate new
  // random trace_id if empty or invalid trace_id was received.
  //
  // This field is required.
  bytes trace_id = 1;
```
Failing to comply will result in failure when exporting the trace. At least the reference collector I've tested with returns an error code.

I'm more of a brutal failure person than the fix gracefully...:)
IMO we should log error and fail to start if single/8-byte identifier is used AND using either of
* w3c context propagation
* OTel exporter

I've seen too many cases where logs are ignored and/or possibly just drown in an avalanche of other logs.
So under the hood start to pad the traceid is IMO not the correct way. If you want 8-byte identifiers then you can't use w3c context nor OTel exporters....period
Better to be clear in the documentation and also fail to start in case you don't read docs.

Reply:
I won't mind if it's added, but it can be merged without it